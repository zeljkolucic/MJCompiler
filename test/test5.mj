program MJProgram

int a, b,c ;
int niz[];

{
	int f1(int x, int y, int z)
	{
		return x + y + z;
	}
	
	int f2(int x)
	{
		return x * 2;
	}
	
	int f3(int x, int y)
	{
		return x - y;
	}

	void main()
	{
		a = f1(f1(1,f2(2),3), f2(f3(10,5)), f3(f1(1,1,1), f2(0)));
		print(a);
	}

}