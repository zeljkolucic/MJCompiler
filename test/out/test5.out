Linija [9]: Pristup formalnom parametru x.
Linija [9]: Pristup formalnom parametru y.
Linija [9]: Pristup formalnom parametru z.
Linija [14]: Pristup formalnom parametru x.
Linija [19]: Pristup formalnom parametru x.
Linija [19]: Pristup formalnom parametru y.
Linija [24]: Pristup globalnoj promenljivoj a.
Linija [24]: Poziv funkcije f2.
Linija [24]: Poziv funkcije f1.
Linija [24]: Poziv funkcije f3.
Linija [24]: Poziv funkcije f2.
Linija [24]: Poziv funkcije f1.
Linija [24]: Poziv funkcije f2.
Linija [24]: Poziv funkcije f3.
Linija [24]: Poziv funkcije f1.
Linija [25]: Pristup globalnoj promenljivoj a.
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog MJProgram: notype, 0, 1 
   Var a: int, 0, 0 
   Var b: int, 1, 0 
   Var c: int, 2, 0 
   Var niz: Arr of int, 3, 0 
   Meth f1: int, 0, 3 
      Var x: int, 0, 1 
      Var y: int, 1, 1 
      Var z: int, 2, 1 

   Meth f2: int, 0, 1 
      Var x: int, 0, 1 

   Meth f3: int, 0, 2 
      Var x: int, 0, 1 
      Var y: int, 1, 1 

   Meth main: notype, 0, 0 



